ui/src/EnvVarForm/components/EnvVarForm.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react'
+import './EnvVarForm.css'
+import Form from 'react-bootstrap/Form'
+import FormGroup from 'react-bootstrap/FormGroup'
+import Button from 'react-bootstrap/Button'
+import APICalls from '../../CustomElements/APICalls'
+import VarForm from '../../CustomElements/VarForm'
+
+class EnvVarForm extends React.Component {
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            envvars: {},
+            items: [],
+            secrets: {},
+        }
+        this.API = new APICalls()
+    }
+
+    async updateVars() {
+        var data = await this.API.APIGet(this.API._CMPATH)
+        this.setState({envvars: data['env']},() => {console.log("Updated EnvVars")})
+        this.renderForms()
+    }
+
+    componentDidMount() {
+        this.updateVars()
+    }
+
+    onBlur(e) {
+        var container = document.getElementById('EnvVarContainer')
+        var vars = {}
+        for (var i = 0; i < container.children.length; i++) {
+            var keyValuePair = container.children[i]
+            var key = keyValuePair.children[0].children[0].children[0] //should be rewritten
+            var value = keyValuePair.children[0].children[1]
+            if (key.value) {
```

Comment:
Verify if this is still needed

Reply:
Changed

Reply:
This could be merged into a single `if` - folllow up issue https://github.com/opendatahub-io/jupyterhub-singleuser-profiles/issues/54